mlflow/data/huggingface_dataset.py

```diff
@@ -0,0 +1,143 @@
+import hashlib
+import json
+import logging
+from functools import cached_property
+from typing import Any, Union, Optional, Mapping, Sequence, Dict
+
+import datasets
+import numpy as np
+import pandas as pd
+
+from mlflow.data.dataset import Dataset
+from mlflow.data.huggingface_dataset_source import HuggingFaceDatasetSource
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+from mlflow.types import Schema
+from mlflow.types.utils import _infer_schema
+
+_logger = logging.getLogger(__name__)
+
+_MAX_ROWS_FOR_DIGEST_COMPUTATION_AND_SCHEMA_INFERENCE = 10000
+
+
+class HuggingFaceDataset(Dataset):
+    """
+    Represents a HuggingFace dataset.
+    """
+
+    def __init__(
+        self,
+        ds: datasets.Dataset,
+        source: HuggingFaceDatasetSource,
+        name: Optional[str] = None,
+        digest: Optional[str] = None,
+    ):
+        self._ds = ds
+        super().__init__(source=source, name=name, digest=digest)
+
+    def _compute_digest(self) -> str:
+        """
+        Computes a digest for the dataset. Called if the user doesn't supply
+        a digest when constructing the dataset.
+        """
+
+        try:
+            df = next(
+                self._ds.to_pandas(
+                    batch_size=_MAX_ROWS_FOR_DIGEST_COMPUTATION_AND_SCHEMA_INFERENCE, batched=True
+                )
+            )
+        except Exception:
+            if hasattr(self._ds, "_fingerprint") and self._ds._fingerprint is not None:
+                return self._ds._fingerprint[:8]
```

Comment:
@prithvikannan `_fingerprint` is not deterministic across dataset downloads / reloads from disk, unfortunately. Conversion of first 10k rows to pandas produces the determinism we're looking for.

Reply:
Removed! :)